src/main/java/com/conveyal/r5/profile/FrequencyRandomOffsets.java

```diff
@@ -44,13 +43,19 @@ public class FrequencyRandomOffsets {
     private final Map<TripSchedule, int[]> offsetsForTripSchedule = new HashMap<>();
 
     /** The mersenne twister is a higher quality random number generator than the one included with Java */
-    private MersenneTwister mt = new MersenneTwister();
+    private MersenneTwister mt;
 
-    public FrequencyRandomOffsets(TransitLayer data) {
+    public FrequencyRandomOffsets(TransitLayer data, ProfileRequest request) {
```

Comment:
Is the entire request necessary to be passed as a parameter? I would recommend passing either the seed itself or a `MersenneTwister`.

Reply:
Assuming we want to allow locking the schedules at all I agree: it would be better to pass in only the seed to deter further coupling with the ProfileRequest. The seed should probably include both lat and lon to avoid correlation between all origins in a row across a grid.

I don't see a clear justification for changing the schedules from one origin to another, but locking them at the per-origin (or per-row) level. This would lead to apparently stable differences between neighboring origins that use the same routes to reach most destinations, which are likely to be misinterpreted. 

The effects of knowingly deriving results from one single permutation of the schedules are clearer if that permutation is the same for all origins. In that case the seed could just be derived from the TransitLayer (e.g. its center point), and the parameter can just be a boolean for whether to do this.